### PR TITLE
MM-14553 Add nps-test command and refactor for test build

### PR DESCRIPTION
**Pull request by hmhealey**
The `nps-test` command will be controlled by a constant, so the test build will have to be manually generated. The other refactoring is just to give me some easy constants when changing some times to allow for QA to test within a reasonable amount of time.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-14553

### CODE REVIEW

File: server/test_commands.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	commands := map[string]func([]string) (*model.CommandResponse, *model.AppError){
+		COMMAND_TEST_RESET:   p.executeTestResetCommand,
+		COMMAND_TEST_VERSION: p.executeTestVersionCommand,
+	}
+
+	command, ok := commands[action]
+	if !ok {
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         "Incorrect action specified.",
```
Comment (hanzei) [inline server/test_commands.go]: What do you think about:

```suggestion
			Text:         fmt.Sprintf("Unkown action %action", command),
```

File: server/test_commands.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+		p.API.LogError("Failed to restore bot user ID after resetting plugin state", "err", err)
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         "Failed to re-add bot user ID after resetting plugin state. This will likely render the plugin inoperable. See log for more details.",
+		}, nil
+	}
+
+	return &model.CommandResponse{
+		ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+		Text:         "NPS plugin reset. Please re-enable it from the system console to continue testing.",
```
Comment (hanzei) [inline server/test_commands.go]:
```suggestion
		Text:         "NPS plugin reset. Please disable and re-enable it from the system console to continue testing.",
```

File: server/test_commands.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+func (p *Plugin) executeTestVersionCommand(args []string) (*model.CommandResponse, *model.AppError) {
+	if len(args) == 0 || len(args) > 2 {
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         "Wrong number of params specified. Expected `" + COMMAND_TEST + " " + COMMAND_TEST_RESET + " <version> [<unix timestamp of upgrade>]`.",
+		}, nil
+	}
+
+	version := semver.MustParse(args[0])
```
Comment (hanzei) [inline server/test_commands.go]: Since this is user input, panicing on a wrong input seams not the right way.

**Comment by hmhealey** [comment]
@hanzei Updated the error handling, and I also removed some logic that's no longer needed with https://github.com/mattermost/mattermost-plugin-nps/pull/18

File: server/test_commands.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+	}
+
+	return nil, nil
+}
+
+func (p *Plugin) executeTestCommand(action string, args []string) (*model.CommandResponse, *model.AppError) {
+	if action == "" {
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         fmt.Sprintf("Unknown action %s specified.", action),
```
Comment (jespino) [inline server/test_commands.go]: The error message is weird, actually you give here the "action" name information when the action is empty, but not some lines bellow where the action is invalid and not empty.

File: server/survey.go [hunk 5]
```diff
@@ -14,16 +14,19 @@ import (
 const (
 	// How often "survey scheduled" emails can be sent to prevent multiple emails from being sent if multiple server
 	// upgrades occur within a short time
-	MIN_DAYS_BETWEEN_SURVEY_EMAILS = 7
+	MIN_TIME_BETWEEN_SURVEY_EMAILS = 7 * 24 * time.Hour
```
Comment (crspeller) [inline server/survey.go]: Not really part of this PR... but shouldn't these be config settings?

File: server/test_commands.go (hunk 4 above)
Comment (hmhealey) [reply server/test_commands.go]: Oops. When addressing Ben's concerns, I accidentally replaced the wrong error message. I'll fix that

File: server/survey.go (hunk 5 above)
Comment (hmhealey) [reply server/survey.go]: I don't think we wanted them to be configurable since it wasn't included in the spec. I don't know if it just wasn't thought about at the time, or if there was some reason for not including them like not letting admins increase the rate that the surveys came out so it would annoy their users.
